packages/@aws-cdk/pipelines/lib/legacy/pipeline.ts

```diff
@@ -242,6 +260,10 @@ export class CdkPipeline extends CoreConstruct {
       throw new Error('You must pass a \'sourceAction\' (or a \'codePipeline\' that already has a Source stage)');
     }
 
+    if (props.enableKeyRotation && props.crossAccountKeys) {
+      throw new Error('Setting \'enableKeyRotation\' to true also requires \'crossAccountKeys\' to be enabled');
+    }
+
```

Comment:
Let's remove this validation from here (it's already done in `Pipeline`):

```suggestion
```

(Also, it's actually wrong - the condition is incorrect. Another great argument for leaving this out 😜)